cranelift/isle/isle/src/codegen.rs

```diff
@@ -534,15 +626,16 @@ impl<'a> Codegen<'a> {
                     match ret_kind {
                         ReturnKind::Plain => write!(ctx.out, "return ")?,
                         ReturnKind::Option => write!(ctx.out, "return Some(")?,
-                        ReturnKind::Iterator => write!(ctx.out, "returns.push(")?,
+                        ReturnKind::Iterator => write!(ctx.out, "returns.extend(Some(")?,
```

Comment:
Is the choice of `Option` for the iterator here significant, or would `std::iter::once` also work?

Reply:
`std::iter::once` would also work, no strong feelings either way.